packages/devtools_app/lib/src/framework/notifications.dart

```diff
@@ -7,84 +7,100 @@ import 'dart:async';
 import 'package:flutter/material.dart';
 import 'package:flutter/scheduler.dart';
 
+import '../primitives/auto_dispose_mixin.dart';
+import '../primitives/notifications.dart';
 import '../primitives/utils.dart';
-import 'common_widgets.dart';
-import 'theme.dart';
-import 'utils.dart';
+import '../shared/common_widgets.dart';
+import '../shared/globals.dart';
+import '../shared/theme.dart';
+import '../shared/utils.dart';
 
 double get _notificationHeight => scaleByFontFactor(175.0);
 final _notificationWidth = _notificationHeight * goldenRatio;
 
-/// Interface for pushing notifications in the app.
-///
-/// Use this interface in controllers that need to show notifications.
-///
-/// Using the interface instead of the [NotificationsState] implementation
-/// will allow you to write unit tests for the controller that consumes it
-/// instead of widget tests.
-abstract class NotificationService {
-  /// Pushes a notification [message].
-  void push(String message);
-}
-
 /// Manager for notifications in the app.
 ///
 /// Must be inside of an [Overlay].
-class Notifications extends StatelessWidget {
-  const Notifications({Key? key, required this.child}) : super(key: key);
+class NotificationsView extends StatelessWidget {
+  const NotificationsView({Key? key, required this.child}) : super(key: key);
 
   final Widget child;
 
-  /// The default duration for notifications to show.
-  static const Duration defaultDuration = Duration(seconds: 7);
-
   @override
   Widget build(BuildContext context) {
     return Overlay(
       initialEntries: [
         OverlayEntry(
-          builder: (context) => _NotificationsProvider(child: child),
+          builder: (context) => _Notifications(child: child),
           maintainState: true,
           opaque: true,
         ),
       ],
     );
   }
-
-  static NotificationsState? of(BuildContext context) {
-    final provider =
-        context.dependOnInheritedWidgetOfExactType<_InheritedNotifications>();
-    return provider?.data;
-  }
 }
 
-class _NotificationsProvider extends StatefulWidget {
-  const _NotificationsProvider({Key? key, required this.child})
-      : super(key: key);
+class NotificationController implements NotificationService {
```

Comment:
it looks like this is the only class that implements NotificationsService. Let's just combine the two classes then, as there isn't any added value by breaking this up into an abstract class and an implementation class. It is also a bit confusing that we have to cast `notificationService as NotificationController` in multiple places.